### DISCUSSION TITLE
view collection issue

### DISCUSSION THREAD

**Discussion by apujayed** [pocketbase/pocketbase#3468] [Q&A]
`SELECT
catalog.Lot_number,
auction_info.price_max
FROM catalog
LEFT JOIN auction_info on auction_info.catalog=catalog.id
GROUP by catalog.id`

why this Invalid query - missing required id column (you can use
'(ROW_NUMBER() OVER as id' if you don't have one). issue occur?

**Comment by ganigeorgiev** [comment]
View collection queries are required to have an `id` column. You can use `ROW_NUMBER` (_or any other SQLite expression_) to generate it dynamically OR use an existing id from your dataset (in your case it could be `catalog.id`):

```sql
SELECT
    catalog.id,
    catalog.Lot_number,
    auction_info.price_max
FROM catalog
LEFT JOIN auction_info on auction_info.catalog=catalog.id
GROUP by catalog.id
```
